src/Search/FluentSearchFormExtension.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * SilverStripe core search work around when using Fluent
+ * @link https://gist.github.com/baukezwaan/266c469758319daea5460f2a5647c525
+ */
+
+namespace TractorCow\Fluent\Search;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\CMS\Search\SearchForm;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\PaginatedList;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * Simple extension to enable SS Core search to work with SS Fluent
+ */
+class FluentSearchForm extends SearchForm
+{
+    /**
+     * Return dataObjectSet of the results using current request to get info from form.
+     * Wraps around {@link searchEngine()}.
+     *
+     * @return PaginatedList
+     */
+    public function getResults()
+    {
+        $keywords = '';
+        $results = new ArrayList();
+        $request = $this->getRequestHandler()->getRequest();
+        $page_start = $request->requestVar('start') ?: 0;
+        $page_length = $this->getPageLength();
+
+        // Define and generate keyword search patterns
+        $keyword_patterns = [
+            [
+                'patterns' => [
+                    '/()("[^()"]+")( and )("[^"()]+")()/i',
+                    '/(^| )([^() ]+)( and )([^ ()]+)( |$)/i',
+                ],
+                'callback' => function ($matches) {
+                    return ' +' . $matches[2] . ' +' . $matches[4] . ' ';
+                },
+            ],
+            [
+                'patterns' => [
+                    '/(^| )( not )("[^"()]+")/i',
+                    '/(^| )( not )([^() ]+)( |$)/i',
+                ],
+                'callback' => function ($matches) {
+                    return ' -' . $matches[3];
+                },
+            ],
+        ];
+        foreach ($keyword_patterns as $k_p) {
+            $keywords = preg_replace_callback(
```

Comment:
you assign $keywords on each loop. This means that only the last loop will be executed (it will overwrite each prior loop).

Assign $keywords = $this->getSearchQuery(); before this loop to fix, and pass in $keywords to the preg_replace_callback()

Reply:
Good spotting, thank you. I've fixed that up to assign the search term before the loop to make sure it replaces find for both 'and' and 'not'.